### ISSUE TITLE
fork/exec  rsync: operation not permitted

### ISSUE THREAD

**Issue by ryjogo** [drakkan/sftpgo#927]
I have enabled rsync for sshd, in the attempt to initiate an rsync connection.. however i get the following when using the docker container... 2.3.1

```
{"level":"error","time":"2022-07-15T08:01:46.440","sender":"SSH","connection_id":"SSH_d22b341582a198b44e0340891b717072c4772ca90392bb2d08d04d8a1b68eb67_1","message":"command failed: \"rs
ync\" args: [--server -vlogDtpre.iLsfxC . /sftp/] user: ryan err: fork/exec /usr/bin/rsync: operation not permitted"}                                                          {"level":"error","time":"2022-07-15T08:01:56.421","sender":"SSH","connection_id":"SSH_d55e44455a6e0cae2adc542a77ccfcb8ef0fb961ddd911e5c7f019356b9e80de_1","message":"command failed: \"rsync\" args: [--server -vvvvlogDtpre.iLsfxC . /sftp/] user: ryan err: fork/exec /usr/bin/rsync: operation not permitted"}
```

rsync seems to be in the container:
```
I have no name!@sftp-6b76f768db-qfq7w:/sftp$ rsync --version
rsync  version 3.2.3  protocol version 31
...
```

verbose on the client says:

```
(Client) Protocol versions: remote=1853453170, negotiated=29
protocol version mismatch -- is your shell clean?
(see the rsync man page for an explanation)
```

**Comment by drakkan** [comment]
It works for me, you probably have some security policy that doesn't allow to start processes within the container.

Please note that the SFTPGo support will change in the near future. Questions like this will remain unanswered. 
I will continue to check the reported issues to see if you are experiencing a bug and if so I will fix it, but will only provide support to project sponsors. For example there is no bug here. Issues like this will remain open with no answer or will be closed as invalid with no answer. I have no more motivation to continue supporting everyone without getting anything in return. Thanks for understanding

**Comment by cuihaoleo** [comment]
I encountered the same issue. It turned out to be because the users has uid / gid mapping set but sftpgo wasn't run with root. It seemed to try to setuid/gid before exec and failed.

I didn't mean to set uid / gid but the users were imported from Linux passwd and inherited the settings. Since WebAdmin hides the setting when sftpgo isn't run with root, I had to modify the sqlite db to reset the setting:
```
$ sudo -u 50000 sqlite3 sftpgo.db
sqlite> update users set uid = 0;
sqlite> update users set gid = 0;
```
